marlowe/src/Language/Marlowe/Client.hs

```diff
@@ -465,15 +472,50 @@ setupMarloweParams owners contract = mapError (review _MarloweError) $ do
                             <> Constraints.mustMintValue (Currency.mintedValue theCurrency)
         let rolesSymbol = Ledger.scriptCurrencySymbol curVali
         let minAdaTxOut = adaValueOf 2
-        let giveToParty (role, pkh) = Constraints.mustPayToPubKey pkh (Val.singleton rolesSymbol role 1 <> minAdaTxOut)
-        let distributeRoleTokens = foldMap giveToParty $ second PaymentPubKeyHash <$> AssocMap.toList owners
+        let giveToParty (role, addr) =
+              mustPayToShelleyAddress addr (Val.singleton rolesSymbol role 1 <> minAdaTxOut)
+        distributeRoleTokens <- foldMapM giveToParty $ AssocMap.toList owners
         let params = marloweParams rolesSymbol
         pure (params, mintTx <> distributeRoleTokens, lookups)
     else do
         let missingRoles = roles `Set.difference` Set.fromList (AssocMap.keys owners)
         let message = T.pack $ "You didn't specify owners of these roles: " <> show missingRoles
         throwing _ContractError $ OtherError message
 
+ownShelleyAddress
```

Comment:
We might consider a PR to move these into `plutus-apps`, but the change would involve adding a constructor to the constraint type there.